addon/index.d.ts

```diff
@@ -0,0 +1,759 @@
+
+  /**
+   * The most recent task instance that errored.
+   */
+  readonly lastErrored: TaskInstance<T> | null;
+
+  /**
+   * The most recently canceled task instance.
+   */
+  readonly lastCanceled: TaskInstance<T> | null;
+
+  /**
+   * The most recent task instance that is incomplete.
+   */
+  readonly lastIncomplete: TaskInstance<T> | null;
+
+  /**
+   * The number of times this task has been performed.
+   */
+  readonly performCount: number;
+
+  /**
+   * Cancels all running or queued `TaskInstance`s for this Task.
+   * If you're trying to cancel a specific TaskInstance (rather
+   * than all of the instances running under this task) call
+   * `.cancel()` on the specific TaskInstance.
+   *
+   * @param options.reason A descriptive reason the task was
+   *   cancelled. Defaults to `".cancelAll() was explicitly called
+   *   on the Task"`.
+   * @param options.resetState If true, will clear the task state
+   *   (`last*` and `performCount` properties will be set to initial
+   *   values). Defaults to false.
+   */
+  cancelAll(options?: { reason?: string, resetState?: boolean }): void;
+
+  /**
+   * Creates a new {@linkcode TaskInstance} and attempts to run it right away.
+   * If running this task instance would increase the task's concurrency
+   * to a number greater than the task's maxConcurrency, this task
+   * instance might be immediately canceled (dropped), or enqueued
+   * to run at later time, after the currently running task(s) have finished.
+   *
+   * @param args Arguments to pass to the task function.
+   */
+  perform(...args: Args): TaskInstance<T>;
+}
+
+/**
+ * "Task Groups" provide a means for applying
+ * task modifiers to groups of tasks. Once a {@linkcode Task} is declared
+ * as part of a group task, modifiers like `drop()` or `restartable()`
+ * will no longer affect the individual `Task`. Instead those
+ * modifiers can be applied to the entire group.
+ *
+ * ```js
+ * import { task, taskGroup } from 'ember-concurrency';
+ *
+ * export default Controller.extend({
+ *   chores: taskGroup().drop(),
+ *
+ *   mowLawn:       task(taskFn).group('chores'),
+ *   doDishes:      task(taskFn).group('chores'),
+ *   changeDiapers: task(taskFn).group('chores')
+ * });
+ * ```
+ */
+export interface TaskGroup<T = unknown> {
+  // readonly isRunning: boolean;
+  // readonly isQueued: boolean;
+  // readonly isIdle: boolean;
+  // readonly state: 'running' | 'queued' | 'idle';
+  // readonly last: TaskInstance<T> | null;
+  // readonly lastRunning: TaskInstance<T> | null;
+  // readonly lastPerformed: TaskInstance<T> | null;
+  // readonly lastSuccessful: TaskInstance<T> | null;
+  // readonly lastComplete: TaskInstance<T> | null;
+  // readonly lastErrored: TaskInstance<T> | null;
+  // readonly lastCanceled: TaskInstance<T> | null;
+  // readonly lastIncomplete: TaskInstance<T> | null;
+  // readonly performCount: number;
+  // cancelAll(options?: { reason?: string, resetState?: boolean }): void;
```

Comment:
I believe these are supposed to be public. However, they are not documented on the public API documentation, so to cautions, I left them out. If they become documented in the future we can add them.

Reply:
I can confirm that they should all be documented/public.

Reply:
I have added (mostly just copied) the docs.